crates/transaction-emitter-lib/src/transaction_generator/mod.rs

```diff
@@ -0,0 +1,46 @@
+// Copyright (c) Aptos
+// SPDX-License-Identifier: Apache-2.0
+
+use aptos_sdk::{
+    move_types::account_address::AccountAddress,
+    transaction_builder::TransactionFactory,
+    types::{transaction::SignedTransaction, LocalAccount},
+};
+use rand::rngs::StdRng;
+use std::{fmt::Debug, sync::Arc};
+
+pub mod account_generator;
+pub mod nft_mint;
+pub mod p2p_transaction_generator;
+
+pub trait TransactionGenerator: Debug + Sync + Send {
+    fn generate_transactions(
+        &mut self,
+        accounts: Vec<&mut LocalAccount>,
+        all_addresses: Arc<Vec<AccountAddress>>,
+        invalid_transaction_ratio: usize,
+        gas_price: u64,
+    ) -> Vec<SignedTransaction>;
+
+    fn gen_single_txn(
+        &self,
+        _from: &mut LocalAccount,
+        _to: &AccountAddress,
+        _num_coins: u64,
+        _txn_factory: &TransactionFactory,
+        _gas_price: u64,
+    ) -> SignedTransaction {
+        unimplemented!()
+    }
+
+    fn generate_invalid_transaction(
+        &self,
+        _rng: &mut StdRng,
+        _sender: &mut LocalAccount,
+        _receiver: &AccountAddress,
+        _gas_price: u64,
+        _reqs: &[SignedTransaction],
+    ) -> SignedTransaction {
+        unimplemented!()
```

Comment:
Why do you have these two methods (gen_single_txn & generate_invalid_transaction) in the parent class, when you don't use them anywhere outside of the classes themselves, and you also don't implement them in all classes?

Reply:
I had kept these because I wanted to add these methods to NFT minting and account creation later but removed them to reduce confusion for now.